addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,611 @@
+    l10n_gr_edi_inv_type = fields.Selection(
+        selection=INVOICE_TYPES_SELECTION,
+        string='MyDATA Invoice Type',
+        compute='_compute_l10n_gr_edi_inv_type',
+        store=True,
+    )
+    l10n_gr_edi_correlation_id = fields.Many2one(
+        comodel_name='account.move',
+        string='MyDATA Correlated Invoice',
+        domain="[('l10n_gr_edi_mark', '!=', False), ('state', '=', 'posted')]",
+        compute='_compute_l10n_gr_edi_correlation_id',
+        store=True,
+    )
+    l10n_gr_edi_payment_method = fields.Selection(
+        selection=PAYMENT_METHOD_SELECTION,
+        string='MyDATA Payment Method',
+        default='1',
+    )
+    l10n_gr_edi_warnings = fields.Json(compute='_compute_l10n_gr_edi_warnings')
+    l10n_gr_edi_need_correlated = fields.Boolean(compute='_compute_l10n_gr_edi_need_fields')
+    l10n_gr_edi_need_payment_method = fields.Boolean(compute='_compute_l10n_gr_edi_need_fields')
+    l10n_gr_edi_enable_send_invoices = fields.Boolean(compute='_compute_l10n_gr_edi_enable_send')
+    l10n_gr_edi_enable_send_expense_classification = fields.Boolean(compute='_compute_l10n_gr_edi_enable_send')
+    l10n_gr_edi_attachment_id = fields.Many2one(comodel_name='ir.attachment')
+
+    ################################################################################
+    # Standard Field Computes
+    ################################################################################
+
+    @api.depends('l10n_gr_edi_document_ids')
+    def _compute_l10n_gr_edi_state(self):
+        for move in self:
+            if gr_documents := move.l10n_gr_edi_document_ids.sorted():
+                move.l10n_gr_edi_state = gr_documents[0].state
+            else:
+                move.l10n_gr_edi_state = False
+
+    @api.depends('l10n_gr_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTENDS 'account'
+        """ Prevent user from resetting the move to draft if it's already sent to MyDATA """
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.l10n_gr_edi_state == 'move_sent':
+                move.show_reset_to_draft_button = False
+
+    @api.depends('country_code', 'state')
+    def _compute_l10n_gr_edi_warnings(self):
+        for move in self:
+            # Warnings are only calculated when the move state is posted
+            if move.country_code == 'GR' and move.state == 'posted':
+                warnings = move._l10n_gr_edi_get_pre_error_dict()
+                move.l10n_gr_edi_warnings = warnings
+            else:
+                move.l10n_gr_edi_warnings = False
+
+    @api.depends('state', 'l10n_gr_edi_state')
+    def _compute_l10n_gr_edi_enable_send(self):
+        for move in self:
+            common_requirements = all((
+                move.country_code == 'GR',
+                move.state == 'posted',
+                move.l10n_gr_edi_state != 'move_sent',
```

Comment:
Maybe a personal taste but I think it's clearer and less difficult to maintain to use a whitelist instead of a blacklist. What about saying explicitely it's for `False` and `move_error`?